[lldb][test] Consolidate libstdc++ and libc++ vector formatter tests into generic test

Michael137 opened the pull request:
The libstdc++ test was a subset of the tests in libc++. This test moves the libc++ test into `generic` and removes the `libstdc++` tests. I retained the `-D_GLIBCXX_DEBUG` test cases though.

Split out from https://github.com/llvm/llvm-project/pull/146740

labath commented on lldb/test/API/functionalities/data-formatter/data-formatter-stl/generic/vector/main.cpp:
```diff
@@ -0,0 +1,40 @@
+
+int main() {
+  int_vect numbers;
+  (numbers.push_back(1));  // break here
```
I'm not sure what are the parens for, but I wouldn't want my name to go down in history with them :P

Michael137 committed:
[lldb][test] Consolidate libstdc++ and libc++ vector formatter tests into generic test

The libstdc++ test was a subset of the tests in libc++. This test moves
the libc++ test into `generic` and removes the `libstdc++` tests. I
retained the `-D_GLIBCXX_DEBUG` test cases though.

Split out from https://github.com/llvm/llvm-project/pull/146740

Michael137 committed:
fixup! remove redundant braces
